core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1885,6 +1885,16 @@ private[spark] class DAGScheduler(
               mapOutputTracker.
                 unregisterMergeResult(shuffleId, reduceId, bmAddress, Option(mapIndex))
             }
+          } else {
+            // Unregister the merge result of <shuffleId, reduceId> if there is a FetchFailed event
+            // and is not a  MetaDataFetchException which is signified by bmAddress being null
+            if (bmAddress != null
+              && bmAddress.executorId.equals(BlockManagerId.SHUFFLE_MERGER_IDENTIFIER)) {
+              assert(pushBasedShuffleEnabled, "Pushed based shuffle needs to " +
+                "be enabled so that merge results are present.")
```

Comment:
`Pushed based shuffle` -> `Push based shuffle`

Change message to something like `Push based shuffle expected to be enabled when handling merge block fetch failure` ? To make it clear what we are expecting the condition to be for.